the-collab-lab/tcl-77-smart-shopping-list
# Issue # 56. Designing responsive list component for  the list page

## Description

Using a combination of React bootstrap, classic Bootstrap, and custom CSS styling to create a responsive layout & design for our lists view page. This PR highlights a history of developers seeking technical decisions that balance requirements, novelty, complexity, and time constraints. The initial decision was to use React Bootstrap's Container, Rol, and Col components to create a  responsive layout. The layout was responsive up until small screen sizes, which debugging revealed to be discrepancies between the column's width and contents. The bug caused the columns to collapse. The responsiveness issues were solved using flexbox in 30 minutes. Flexbox avoided unnecessary complexity tfor the purposes of this issue. 

Our team also agreed that including the add item form and share list form into our list page view improved our user experience so to reduce the necessary number of user clicks to access those features.  

Figma design 
https://www.figma.com/design/v0vHRdn5Gq8kNJSepE5xM8/TCL--GrocerEase?node-id=0-1&node-type=canvas&t=Xf7D0PzYQFTNu6Ga-0


## Related Issue

closes #56 

## Acceptance Criteria

- [ ] Using the teams' figma design, work closely to align the design styles
- [ ] Embed the add item and share list forms inside the list page view
- [ ] The list page is responsive across all screen sizes, especially mobile devices. 

## Type of Changes

`enhancement`

## Updates

### Before

<img width="1050" alt="before design" src="https://github.com/user-attachments/assets/f4b23f90-9bea-4f13-a9d8-ece67feb8638">
<img width="723" alt="beforeurgencystatus" src="https://github.com/user-attachments/assets/f80a8001-411d-4433-8355-eecbafddb05f">

https://github.com/user-attachments/assets/03decd1d-18f0-47fe-a8cc-e85604103890



### After


https://github.com/user-attachments/assets/01657564-44fd-437e-a4f6-396938787a3b


https://github.com/user-attachments/assets/d63bf543-8cbb-44d3-8106-6c02e18e1e98



## Testing Steps / QA Criteria

1. Make sure you are signed into the application
2. Navigate to a list from the home view page
3. Compare and contrast the design with figma design
4. Adjust the viewport width of the webpage to test the responsiveness (Google Chrome's responsiveness, etc)
